napari/_qt/layer_controls/qt_layer_controls_base.py

```diff
@@ -106,3 +108,9 @@ def _on_blending_change(self, event=None):
                 self.layer.blending, Qt.MatchFixedString
             )
             self.blendComboBox.setCurrentIndex(index)
+
+    def close(self):
+        """Layer widget is closing."""
+        super().close()
```

Comment:
couple minor questions I have (rarely, but sometimes) run into tricky issues when re-implementing `close`:
- do you have any thoughts on calling `super().close()` _before_ the other things?  I would have assumed at the end (though a race condition would seem rare here).
- did you have any thoughts on avoiding reimplementing close altogether, and just using the [`self.destroyed`](https://doc.qt.io/qt-5/qobject.html#destroyed) signal to connect `disconnect_events` along with the [`WA_DeleteOnClose`](https://doc.qt.io/qt-5/qt.html#WidgetAttribute-enum) flag?

Reply:
yeah, this may be critical. tests look like they are failing now due to race conditions ...... will give this a go

Reply:
> did you have any thoughts on avoiding reimplementing close altogether, and just using the self.destroyed signal to connect disconnect_events along with the WA_DeleteOnClose flag, (as mentioned in #1984 (comment))?

I've tried this now locally but am still running into problems, with destroyed signal not getting emitted and my `_disconnect_events` not getting called - might relate to this https://stackoverflow.com/questions/16842955/widgets-destroyed-signal-is-not-fired-pyqt

Any ideas @tlambert03?

Reply:
here is relevant diff https://github.com/napari/napari/pull/1984/commits/d899d8cd24704ed5caf8ffa205c17125aecf1a56

Reply:
i've gone back to the other approach but made sure all the `close` calls happen at the end, see https://github.com/napari/napari/pull/1984/commits/3c01f91a8029fec65f13acf29108039a65d479a2. Let's see if tests pass!!